.github/workflows/.reusable-subgraph-workflow.yaml

```diff
@@ -0,0 +1,40 @@
+name: DEPLOY SUBGRAPH
+
+on:
+  workflow_call:
+    inputs:
+      NETWORK_NAMES:
+        required: true
+        type: string
+      SUBGRAPH_FOLDER_NAME: 
+        required: true
+        type: string
+    secrets:
+      SUBGRAPH_ACCESS_TOKEN:
+        required: true
+env: 
+  NETWORK_NAMES: ${{ inputs.NETWORK_NAMES }}
+  SUBGRAPH_FOLDER_NAME: ${{ inputs.SUBGRAPH_FOLDER_NAME }}
+  SUBGRAPH_ACCESS_TOKEN: ${{ secrets.SUBGRAPH_ACCESS_TOKEN }}
+
+jobs: 
+  deploy:
+    name: Deploy
+    runs-on: self-hosted
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Build and Deploy
+        if: ${{ github.ref == 'refs/heads/dev' }}
```

Comment:
Let's have the step for production too - I know doesn't currently exist. This is a sample you can follow: https://github.com/warpfinance/subgraphs/blob/dev/.github/workflows/v2-supported.yaml

Reply:
done